tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -0,0 +1,195 @@
+            if r == 0:
+                return None
+            if r == 1:
+                return 64  # practical minimum
+            return math.floor(2**(self.rnd.random() * 35 + 5))  # up to 1 TB/s
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_BPS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 30))  # up to 1 GB/s
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_RATIO:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            if r == 1:
+                return 1
+            return self.rnd.random()
+
+        if prop == self.ConfigProp.THROTTLE_DELAY_MAX_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 25))  # up to ~1 year
+
+        if prop == self.ConfigProp.BAL_WINDOW_MS:
+            r = self.rnd.randrange(4)
+            if r == 0:
+                return 1
+            if r == 1:
+                return 2147483647
+            return math.floor(2**(self.rnd.random() * 31))
+
+        if prop == self.ConfigProp.BAL_PERIOD_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 22))  # up to ~1.5 months
+
+        raise Exception(f"Unsupported ConfigProp: {prop}")
+
+    def setUp(self):
+
+        self.config = {}
+        for prop in list(self.ConfigProp):
+            val = self.get_config_parameter_random_value(prop)
+            self.config[prop] = val
+            if not val is None:
+                self.redpanda.add_extra_rp_conf({prop.value: val})
+
+        self.logger.info(
+            f"Initial cluster props: {self.redpanda._extra_rp_conf}")
+        super(ThroughputLimitsSnc, self).setUp()
+
+    def current_effective_node_quota(self) -> Tuple[int, int]:
+        metrics = self.redpanda.metrics_sample(
+            "quotas_snc_quota_effective",
+            metrics_endpoint=MetricsEndpoint.METRICS)
+
+        assert metrics, "Effecive quota metric is missing"
+        self.logger.debug(f"Samples: {metrics.samples}")
+
+        node_quota_in = sum(
+            int(s.value) for s in metrics.label_filter({
+                "direction": "ingress"
+            }).samples)
+        node_quota_eg = sum(
+            int(s.value) for s in metrics.label_filter({
+                "direction": "egress"
+            }).samples)
+        return node_quota_in, node_quota_eg
+
+    @cluster(num_nodes=3)
+    def test_configuration(self):
+        """
+        Test various configuration patterns, including extreme ones,
+        verify that it does not wreck havoc onto cluster
+        """
+
+        # TBD: parameterize to run under load or not
+
+        errors = []
+        # 12 passes approximately take 30 secs on a 3-node cluster
+        for _ in range(12):
```

Comment:
nit: you can do something like

```python
@parameterize(count=12)
@cluster(num_nodes=3)
def test_configuration(self, count):
   ...
```

Reply:
what is the advantage of doing it this way?

Reply:
then a key parameter of your test is not mixed in with the rest of your test code and it's easy to adjust. it's not important other than being convenient for folks to change later.

Reply:
no need to chnage if you think it is not important.